bin/init.sh

```diff
@@ -42,55 +42,94 @@ if [ ${ROOT} != "${GO_TOP:-$HOME/go}/src/istio.io/istio" ]; then
        exit 1
 fi
 
+# Get the debug istio.io/proxy docker image. We'll use this image to extract the envoy executable used by
+# some of our tests.
+PROXY_DOCKER_IMAGE=$(grep envoy-debug pilot/docker/Dockerfile.proxy_debug  |cut -d ' ' -f2)
 PROXYVERSION=$(grep envoy-debug pilot/docker/Dockerfile.proxy_debug  |cut -d: -f2)
-PROXY=debug-$PROXYVERSION
+PROXY_EXTRACTED_BIN_NAME=envoy-$PROXYVERSION
 
-# Save envoy in $out
-if [ ! -f $OUT/envoy-$PROXYVERSION ] ; then
+if [ ! -f $OUT/$PROXY_EXTRACTED_BIN_NAME ] ; then
     mkdir -p $OUT
     pushd $OUT
-    # New version of envoy downloaded. Save it to cache, and clean any old version.
 
-    DOWNLOAD_COMMAND=""
-    if command -v curl > /dev/null; then
-       if curl --version | grep Protocols  | grep https > /dev/null; then
-	   DOWNLOAD_COMMAND='curl -Lso -'
-       else
-           echo curl does not support https, will try wget for downloading files.
-       fi
-    else
-       echo curl is not installed, will try wget for downloading files.
-    fi
-
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        if command -v wget > /dev/null; then
-	    DOWNLOAD_COMMAND='wget -qO -'
-        else
-            echo wget is not installed.
-        fi
-    fi
-
-    if [ -z "${DOWNLOAD_COMMAND}" ]; then
-        echo Error: curl is not installed or does not support https, wget is not installed. \
-             Cannot download envoy. Please install wget or add support of https to curl.
-        exit 1
-    fi
-
-    echo "Downloading envoy $PROXY using $DOWNLOAD_COMMAND"
-    time ${DOWNLOAD_COMMAND} https://storage.googleapis.com/istio-build/proxy/envoy-$PROXY.tar.gz | tar xz
-    cp usr/local/bin/envoy $OUT/envoy-$PROXYVERSION
+    # Pull the image to the local docker registry.
+    DOWNLOAD_COMMAND="docker pull $PROXY_DOCKER_IMAGE"
+    echo "Downloading envoy using $DOWNLOAD_COMMAND"
+    ${DOWNLOAD_COMMAND}
+
+    # Run the image and pull out the envoy executable
+    ENVOY_EXTRACT_COMMAND="docker run $PROXY_DOCKER_IMAGE cat /usr/local/bin/envoy > $PROXY_EXTRACTED_BIN_NAME"
```

Comment:
please be mindful, this ends up killing prow by cat'ing large binary into the log
https://storage.googleapis.com/istio-prow/pull/istio_istio/3690/istio-presubmit/5550/build-log.txt